{
  RuntimeException failure=new RuntimeException("FAILURE");
  Iterable<DataFactory<InputNode>> data=DataFactories.nodeData(CsvInputTest.<InputNode>data(":ID,name\n1,Mattias",new FailingNodeDecorator(failure)));
  Input input=new CsvInput(data,defaultFormatNodeFileHeader(),null,null,IdType.INTEGER,COMMAS,badRelationships(0));
  try (InputIterator<InputNode> nodes=input.nodes().iterator()){
    nodes.next();
  }
 catch (  RuntimeException e) {
    assertTrue(e == failure);
  }
}
